test/main.test.js

```diff
@@ -129,4 +130,50 @@ describe('LoopBack Context', function() {
       ], done);
     });
   });
+  it('doesn\'t mix up contexts if using concurrently then() from when 3.7.7',
+  function() {
+    expect(require('when/package.json').version).to.equal('3.7.7');
+    var timeout = 50;
+    // Concurrent execution number 1 of 2
+    var execution1 = new Promise(function execution1(outerResolve, reject) {
+      LoopBackContext.runInContext(function pushToContext1() {
```

Comment:
Whoops, you are right, they are not invoking the middleware. The tests pass anyway (and a test app with concurrent requests works), but that's just because the middleware factory was called before, somewhere... I'll need to fix that.